### PR TITLE
fix(scripts): serve grpc for a web viewer

### LINKED ISSUE #2898: Rerun can not display datasets in the distant mode
### Ticket Type

🐛 Bug Report (Something isn't working)

### Environment & System Info

```Shell
- LeRobot version: 0.4.2
- Platform: Windows-10-10.0.26200-SP0
- Python version: 3.10.19
- Huggingface Hub version: 0.35.3
- Datasets version: 4.1.1
- Numpy version: 2.2.6
- FFmpeg version: N/A
- PyTorch version: 2.7.1+cpu
- Is PyTorch built with CUDA support?: False
- Cuda version: N/A
- GPU model: N/A
- Using GPU in script?: false
- lerobot scripts: ['lerobot-calibrate', 'lerobot-dataset-viz', 'lerobot-edit-dataset', 'lerobot-eval', 'lerobot-find-cameras', 'lerobot-find-joint-limits', 'lerobot-find-port', 'lerobot-imgtransform-viz', 'lerobot-info', 'lerobot-record', 'lerobot-replay', 'lerobot-setup-motors', 'lerobot-teleoperate', 'lerobot-train']
```

### Description

When I accessed to the rerun web viewer, the following error was occurred.

> Data source rerun+http://localhost:9876/proxy has left unexpectedly: gRPC error, message: "js api error: TypeError: NetworkError when attempting to fetch resource."

### Context & Reproduction

Run the following command:

```
lerobot-dataset-viz --repo-id "lerobot/pusht"  --episode-index 0 --mode distant --display-compressed-images true
```

and then access to the web viewer: http://localhost:9090/?url=rerun%2Bhttp%3A%2F%2Flocalhost%3A9876%2Fproxy

### Relevant logs or stack trace

```Shell
No error messages in the script.
```

### Checklist

- [x] I have searched existing tickets to ensure this isn't a duplicate.
- [x] I am using the latest version of the `main` branch.
- [x] I have verified this is not an environment-specific problem.

### Additional Info / Workarounds

When you start the web viewer, you should provide the URI of a gRPC server.

Reference in rerun code:

https://github.com/rerun-io/rerun/blob/383d37db68ae92a16eaf2f669090db1142e39ef2/rerun_py/rerun_sdk/rerun/web.py#L13C1-L18C8

### PR DESCRIPTION
**Pull request by sotanakamura**
## Type / Scope

- **Type**: Bug
- **Scope**: lerobot_dataset_viz.py

## Summary / Motivation

Serving grpc is necessary to visualize data in the distant mode.

## Related issues

fix #2898 

## What changed

Serve grpc and pass its uri to the web viewer.

## How was this tested (or how to run locally)

Run the lerobot_dataset_viz as follows:

```sh
lerobot-dataset-viz --repo-id "lerobot/pusht"  --episode-index 0 --mode distant --display-compressed-images true
```
Access to http://localhost:9090/?url=rerun%2Bhttp%3A%2F%2Flocalhost%3A9876%2Fproxy

## Checklist (required before merge)

- [x] Linting/formatting run (`pre-commit run -a`)
- [x] All tests pass locally (`pytest`)
- [x] Documentation updated
- [x] CI is green

## Reviewer notes

- Anything the reviewer should focus on (performance, edge-cases, specific files) or general notes.
- Anyone in the community is free to review the PR.

### CODE REVIEW

**Comment by CarolinePascal** [comment]
Hi @sotanakamura,

We (@imstevenpmwork) refactored our rerun visualizer to correctly support distant viewers (fbe4c8b94fdb1d62773e0f285f75aebb5b9ec27d). The implementation differs from your PR but it should solve the issue. I'm closing this PR, but feel free to open a new one if you spot another issue !

Best,

Caroline.

**Comment by sotanakamura** [comment]
This problem has not been solved yet in the latest code. Please run the command to reproduce the bug.

```
lerobot-dataset-viz --repo-id "lerobot/pusht"  --episode-index 0 --mode distant --display-compressed-images true
```

Access to http://localhost:9090/?url=rerun%2Bhttp%3A%2F%2Flocalhost%3A9876%2Fproxy

**Comment by CarolinePascal** [comment]
You're right, sorry ! I re-opened the PR and will try it asap - Thanks for notifying me.

**Comment by CarolinePascal** [comment]
Hi @sotanakamura,

This PR indeed solves the issue, again sorry for the early deletion. Before merging it, could you change `open_browser` to `True` (it's handier than having to get the URL manually) ? Thank you !

Best,

Caroline.

**Comment by CarolinePascal** [comment]
Duplicate of https://github.com/huggingface/lerobot/pull/2534

**Comment by sotanakamura** [comment]
Thank you for reviewing my PR. I found that opening a web viewer in a remote machine dose not work well. Instead of that, I added command line output to show the web viewer URI.

I also fixed the following problems:

* fix the command line help
* deprecate the unused ws_port option
* remove "required" from --display-compressed-images
* add grpc_port option

**Comment by CarolinePascal** [comment]
Thank you for the additions ! Deprecating websockets was the next item in my TODO 👍

Though I'm not 100% sure the way you recover the distant machine IP will be robust on all setups... I think you can safely ask the user to provide the distant machine IP in the URL, or to use ssh forwarding as we did for websockets (c.f. file header). 

Also to avoid syntax errors on the URL, you can stick to the `rerun` command : `rerun rerun+http://{remote_ip}:{grpc_port}/proxy`.

**Comment by CarolinePascal** [comment]
Quick follow up @sotanakamura - Do you think you will have time to make the requested modifications ?

**Comment by sotanakamura** [comment]
I can work on this tomorrow.

**Comment by sotanakamura** [comment]
@CarolinePascal Thank you for your patient. I removed the ip detection mechanism and add the output to show the following command: `rerun rerun+http://IP:{grpc_port}/proxy`. When I used port forwarding, I could not open rerun.

I tested this in a local network, so it may not work properly through the internet.

File: src/lerobot/scripts/lerobot_dataset_viz.py [hunk 1]
```diff
@@ -130,19 +128,10 @@ def visualize_dataset(
     gc.collect()
 
     if mode == "distant":
-        server_uri = rr.serve_grpc(grpc_port=grpc_port)
+        server_uri = rr.serve_grpc()
```
Comment (CarolinePascal) [inline src/lerobot/scripts/lerobot_dataset_viz.py]: `grpc_port` is missing !

Comment (sotanakamura) [reply src/lerobot/scripts/lerobot_dataset_viz.py]: I fixed it.

**Review by CarolinePascal — APPROVED** [review]
All good, let's merge !

File: src/lerobot/scripts/lerobot_dataset_viz.py [hunk 2]
```diff
@@ -266,8 +273,7 @@ def main():
     parser.add_argument(
         "--display-compressed-images",
         type=bool,
```
Comment (tc-huang) [inline src/lerobot/scripts/lerobot_dataset_viz.py]: Remove `type=bool` because it cannot be used with `action="store_true"`.
```suggestion

```

File: src/lerobot/scripts/lerobot_dataset_viz.py [hunk 3]
```diff
@@ -266,8 +273,7 @@ def main():
...
-        required=True,
-        default=False,
+        default=True,
```
Comment (tc-huang) [inline src/lerobot/scripts/lerobot_dataset_viz.py]: Use `action="store_true"`. The argument is `False` by default and `True` only when explicitly set.
```suggestion
        action="store_true",
```

**Review by CarolinePascal — APPROVED** [review]
